projects/schematics/src/shared/utils/file-utils.ts

```diff
@@ -126,33 +140,264 @@ export function defineProperty(
+  if (params.length !== parameterClassTypes.length) {
+    return false;
+  }
+
+  return true;
+}
+
+export function addConstructorParam(
+  source: ts.SourceFile,
+  sourcePath: string,
+  constructorNode: ts.Node | undefined,
+  paramToAdd: ClassType
+): Change[] {
+  if (!constructorNode) {
+    throw new SchematicsException(`No constructor found in ${sourcePath}.`);
+  }
+
+  const changes: Change[] = [];
+
+  changes.push(
+    injectService(
+      constructorNode,
+      sourcePath,
+      paramToAdd.className,
+      'no-modifier'
+    )
+  );
+
+  if (!isImported(source, paramToAdd.className, paramToAdd.importPath)) {
+    changes.push(
+      insertImport(
+        source,
+        sourcePath,
+        paramToAdd.className,
+        paramToAdd.importPath
+      )
+    );
+  }
+
+  changes.push(
+    updateConstructorSuperNode(
+      sourcePath,
+      constructorNode,
+      paramToAdd.className
+    )
+  );
+
+  return changes;
+}
+
+function updateConstructorSuperNode(
+  sourcePath: string,
+  constructorNode: ts.Node,
+  propertyName: string
+): InsertChange {
+  const callExpressions = findNodes(
+    constructorNode,
+    ts.SyntaxKind.CallExpression
+  );
+  propertyName = strings.camelize(propertyName);
+
+  if (callExpressions.length === 0) {
+    throw new SchematicsException('No super() call found.');
+  }
+  // super has to be the first expression in constructor
+  const firstCallExpression = callExpressions[0];
+  const superKeyword = findNodes(
+    firstCallExpression,
+    ts.SyntaxKind.SuperKeyword
+  );
+  if (superKeyword && superKeyword.length === 0) {
+    throw new SchematicsException('No super() call found.');
+  }
+
+  let toAdd = '';
+  let position: number;
+
+  const params = findNodes(firstCallExpression, ts.SyntaxKind.Identifier);
+  // just an empty super() call, without any params passed to it
+  if (params && params.length === 0) {
+    position = superKeyword[0].end + 1;
+  } else {
+    const lastParam = params[params.length - 1];
+    toAdd += ', ';
+    position = lastParam.end;
+  }
+
+  toAdd += propertyName;
+  return new InsertChange(sourcePath, position, toAdd);
+}
+
 export function injectService(
-  nodes: ts.Node[],
+  constructorNode: ts.Node | undefined,
   path: string,
   serviceName: string,
+  modifier: 'private' | 'protected' | 'public' | 'no-modifier',
   propertyName?: string
 ): InsertChange {
-  const constructorNode = nodes.find(n => n.kind === ts.SyntaxKind.Constructor);
-
   if (!constructorNode) {
     throw new SchematicsException(`No constructor found in ${path}.`);
   }
 
-  const parameterListNode = constructorNode
-    .getChildren()
-    .find(n => n.kind === ts.SyntaxKind.SyntaxList);
+  const constructorParameters = findNodes(
+    constructorNode,
+    ts.SyntaxKind.Parameter
+  );
 
-  if (!parameterListNode) {
-    throw new SchematicsException(
-      `No no parameter list found in ${path}'s constructor.`
-    );
+  let toAdd = '';
```

Comment:
Can we change it to a more descriptive name?

Reply:
The same as above